src/internal.c

```diff
@@ -33629,24 +33542,24 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
         InternalTicket* it;
         int             ret;
         int             outLen;
-        word32          inLen;
+        word16          inLen;
 
         WOLFSSL_START(WC_FUNC_TICKET_DO);
         WOLFSSL_ENTER("DoClientTicket");
 
         if (len > SESSION_TICKET_LEN ||
-             len < (word32)(sizeof(InternalTicket) + WOLFSSL_TICKET_FIXED_SZ)) {
+            len < (word32)(sizeof(InternalTicket) + WOLFSSL_TICKET_FIXED_SZ)) {
             WOLFSSL_ERROR_VERBOSE(BAD_TICKET_MSG_SZ);
-            return BAD_TICKET_MSG_SZ;
+            return WOLFSSL_TICKET_RET_REJECT;
         }
 
         et = (ExternalTicket*)input;
 
         /* decrypt */
-        ato32(et->enc_len, &inLen);
-        if (inLen > (word16)(len - WOLFSSL_TICKET_FIXED_SZ)) {
+        ato16(et->enc_len, &inLen);
```

Comment:
Please revert back to using 32-bit here! This was specifically changed to 32-bit to avoid an alignment padding issue. See https://github.com/wolfSSL/wolfssl/pull/4908

Reply:
Also please add an inline code comment to prevent this from changing again in the future. The structure element `enc_len` needs to be 32-bit aligned.

Reply:
From @julek-wolfssl : 

I disagree that this (referring to your last comment in 5476) needs to be reverted. I took a look at the original issue and the cause of the error was that in 5.2 the InternalTicket looked like this:

```
    /* our ticket format */
    typedef struct InternalTicket {
        ProtocolVersion pv;                    /* version when ticket created */
        byte            suite[SUITE_LEN];      /* cipher suite when created */
        byte            msecret[SECRET_LEN];   /* master secret */
        word32          timestamp;             /* born on */
        word16          haveEMS;               /* have extended master secret */
#ifdef WOLFSSL_TLS13
        word32          ageAdd;                /* Obfuscation of age */
        word16          namedGroup;            /* Named group used */
        TicketNonce     ticketNonce;           /* Ticket nonce */
    #ifdef WOLFSSL_EARLY_DATA
        word32          maxEarlyDataSz;        /* Max size of early data */
    #endif
#endif
    } InternalTicket;
```

It has word32 and word16 members. I fixed this in 617eda9d449a3755e9791ae8a8ce26ff43a7304b by changing everything to byte buffers. I ran into the same issue described in 4887 and changing the struct over to byte arrays was the correct fix.

Reply:
FYI: `enc_len` used to be `word16`, then we changed it to `word32` and now changing it back to `word16`.